includes/events-functions.php

```diff
@@ -0,0 +1,255 @@
+<?php
+/**
+ * Functions related to event feed retrieval and display
+ * in the This Week/end @ UCF emails
+ */
+namespace GMUCF\Theme\Includes\Events;
+use GMUCF\Theme\Includes\Utilities;
+
+
+/**
+ * Compare events based on start time
+ *
+ * @return integer
+ * @author Chris Conover
+ **/
+function compare_event_starts($a, $b) {
+	// Just take into account the time part of the start date time.
+	// Otherwise events that are ongoing would be put in front.
+	$a_parts = explode(' ', $a->starts);
+	$b_parts = explode(' ', $b->starts);
+	$a_start = strtotime($a_parts[4]);
+	$b_start = strtotime($b_parts[4]);
+	if($a_start == $b_start) {
+		return 0;
+	} else {
+		return ($a_start > $b_start) ? +1 : -1;
+	}
+}
+
+
+/**
+ * Fetch events from UCF event system (with caching)
+ *
+ * @return array
+ * @author Chris Conover
+ **/
+function get_event_data($options = array()) {
+	$cache_key_prefix = 'events-';
+	$default_options = array(
+		'calendar_id' => get_option( 'events_calendar_id' ),
+		'per_page'    => get_option( 'events_limit' ),
+		'format'      => 'json');
+
+	$options = array_merge($default_options, $options);
+
+	$cache_key = $cache_key_prefix.implode('', $options);
+
+	if(CLEAR_CACHE || ($events = get_transient($cache_key)) === False) {
+		$events = array();
+
+		$url = get_option( 'events_url' ) . $options['y'].'/'.$options['m'].'/'.$options['d'].'/feed.json';
+
+		$params = array(
+			'per_page' => $options['per_page']
+		);
+
+		$url .= '?' . http_build_query( $params );
+
+		$ch = curl_init($url);
+		curl_setopt($ch, CURLOPT_AUTOREFERER, TRUE);
+		curl_setopt($ch, CURLOPT_HEADER, 0);
+		curl_setopt($ch, CURLOPT_TIMEOUT, get_option( 'events_timeout' ));
+		curl_setopt($ch, CURLOPT_RETURNTRANSFER, 1);
+		curl_setopt($ch, CURLOPT_FOLLOWLOCATION, TRUE);
+
```

Comment:
We need to add an issue to replace this with the built in WordPress function.

Reply:
Done. https://github.com/UCF/UCF-GMUCF-Theme/issues/68